irc/client.d

```diff
@@ -241,12 +247,40 @@ class IrcClient
 		socket.send("\r\n"); // TODO: should be in one call to send?
 	}
 
+	enum additionalMsgLens{
+		PRIVMSG=MAX_USERHOST_LEN,
+		NOTICE=MAX_USERHOST_LEN
+	}
+	
+	/**
+	 * Returns the additional lengthrequirements of a method.
+	 * 
+	 * Params:
+	 * 	method = The method to query.
+	 * Returns:
+	 * 	The additional lengthrequirements.
+	 */
+	private static uint additionalMsgLen(string method)(){
```

Comment:
This can just be a template, no need for a function:
```d
template additionalMsgLen(string method)
{
    static if(staticIndexOf!(method, __traits(allMembers, AdditionalMsgLens)) == -1)
        return 0;
    else
        enum additionalMsgLen = mixin("additionalMsgLens." ~ method);
}
```
The tests can be static asserts:
```d
static assert(additionalMsgLen!"PRIVMSG" == MAX_USERHOST_LEN);
...
```